dbms/src/Flash/Planner/PhysicalPlan.cpp

```diff
@@ -93,6 +93,11 @@ void PhysicalPlan::build(const tipb::Executor * executor)
 {
     RUNTIME_CHECK(executor);
     RUNTIME_CHECK(executor->has_executor_id());
+    // For plans generated by TiDB, assume 1st hashagg can only be followed by ExchangeSender.
+    // So only ExchangeSender can followed by an auto passthrough hashagg.
+    if (executor->tp() != tipb::ExecType::TypeExchangeSender)
```

Comment:
todo: As discussed with @windtalker , delete this check in planner, add check in Operator